test/state/mpt.cpp

```diff
@@ -128,16 +141,15 @@ class MPTNode
         br.m_children[idx1] = std::move(child1);
         br.m_children[idx2] = std::move(child2);
 
-        return (path.length != 0) ? ext(path, std::make_unique<MPTNode>(std::move(br))) :
-                                    std::move(br);
+        return (!path.empty()) ? ext(path, std::make_unique<MPTNode>(std::move(br))) :
+                                 std::move(br);
     }
 
     /// Finds the position at witch two paths differ.
     static size_t mismatch(const Path& p1, const Path& p2) noexcept
     {
-        assert(p1.length <= p2.length);
-        return static_cast<size_t>(
-            std::mismatch(p1.nibbles, p1.nibbles + p1.length, p2.nibbles).first - p1.nibbles);
+        assert(p1.size() <= p2.size());
+        return static_cast<size_t>(std::ranges::mismatch(p1, p2).in1 - p1.begin());
```

Comment:
why is `std::ranges::mismatch()` better?

Reply:
You don't have to put a pair of iterators. It handles the case when the second range is shorter. And outputs both iterators.

Reply:
`std::mismatch()` also handles shorter second range in C++20 required by `std::range`, and outputs a pair of iterators, so the only convenience is passing containers instead of iterators, it seems.

Reply:
Yes, correct. The range addition is for convenience and readability.